tests/test_authz_authorizer.py

```diff
@@ -38,37 +32,161 @@ def test__init__(self):
+        authz = self._mocked_make_authorizer({**self.token_payload, "allow": {ALL: None}})
+        with pytest.raises(PermissionDenied):
+            authz.check_access()
+
+    def test__check_resource(self) -> None:
+        with patch.object(self.authz, "_deny_if_all", new_callable=MagicMock()) as mocked_deny:
+            self.authz._check_resource({"res": "xxx"})  # pylint: disable=protected-access
+            mocked_deny.assert_has_calls(
+                [
+                    call({"res": "xxx"}),
+                    call("res"),
+                    call("xxx"),
+                ]
+            )
+
+    def test_validate_allow_all(self) -> None:
+        self.authz.check_access()
+        assert self.authz.outcome == ALLOW
+        assert self.authz.restrictions == {"allow": "*", "deny": None}
+
+    def test_check_allow_fail_all(self) -> None:
+        self.authz.allow = None
+        with pytest.raises(PermissionDenied):
+            self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+            assert self.authz.outcome == ALLOW
+
+    def test_check_allow_domain(self) -> None:
+        self.authz.allow = {"test_resource": ALL}
+        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert self.authz.outcome == ALLOW
+
+    def test_check_allow_one(self) -> None:
+        self.authz.allow = {"test_resource": {"permission_name": {"allow": ALL}}}
+        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert self.authz.outcome == ALLOW
+
+    def test_check_allow_one_scope(self) -> None:
+        self.authz.allow = {"test_resource": {"permission_name": {"allow": "self"}}}
+        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert self.authz.outcome == ALLOW
+        assert self.authz.allowed_resource == "self"
+
+    def test_check_allow_fail_one_scope(self) -> None:
+        self.authz.allow = {"test_resource": {"permission_name": {"deny": "self"}}}
+        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert self.authz.outcome == ALLOW
+        assert self.authz.denied_resource == "self"
+
+    def test_get_restrictions_restricted(self) -> None:
+        self.authz.allowed_resource = ALL
+        self.authz.denied_resource = {"city": "warszawa"}
+        assert self.authz.restrictions == {"allow": ALL, "deny": {"city": "warszawa"}}
+
+    def test_check_deny_one_scope(self) -> None:
+        self.authz.deny = {"test_resource": {"permission_name": "self"}}
+        self.authz._check_deny()  # pylint: disable=protected-access
+        assert self.authz.denied_resource == "self"
+
+    def test__set_policy_w_scope(self) -> None:
         self.authz._set_policy(  # pylint: disable=protected-access
-            "", {"allow": "Lambda", "deny": "Lambda"}
+            "",
+            {
+                **self.token_payload,
```

Comment:
Why do you need `self.token_payload` for `policy_override` - this test worked before 🤔